src/net_processing.cpp

```diff
@@ -495,14 +502,16 @@ class PeerManagerImpl final : public PeerManager
      * @return Returns true if the peer was punished (probably disconnected)
      */
     bool MaybePunishNodeForBlock(NodeId nodeid, const BlockValidationState& state,
-                                 bool via_compact_block, const std::string& message = "");
+                                 bool via_compact_block, const std::string& message = "")
+        EXCLUSIVE_LOCKS_REQUIRED(!m_peer_mutex);
```

Comment:
Same here, as it would be better to just pass the `Peer&`

Reply:
Tried this, but didn't work, as the Peer might be non-existent (disconnected) at this point.